### PR TITLE
Add ability to co-install dev build with prod

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
Tested by installing Daimo Dev app with current Testflight/Play Store and using them simultaneously successfully on iOS and Android.

Have not run/tested a fresh prod build, but maybe we should just do that during release testing since it'll increment versions.

### CODE REVIEW

File: apps/daimo-mobile/eas.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
       },
       "env": {
         "DAIMO_DOMAIN": "daimo.xyz",
-        "DAIMO_PASSKEY_DOMAIN": "daimo.xyz"
+        "DAIMO_PASSKEY_DOMAIN": "daimo.xyz",
+        "APP_VARIANT": "sim"
```
Comment (dcposch) [inline apps/daimo-mobile/eas.json:16]: nit, `DAIMO_APP_VARIANT` ?

Comment (nalinbhardwaj) [reply apps/daimo-mobile/eas.json]: done

File: apps/daimo-web/public/.well-known/apple-app-site-association [hunk 2]
```diff
@@ -17,12 +18,14 @@
...
-      "JV8PYC9QV4.com.daimo"
+      "JV8PYC9QV4.com.daimo",
+      "JV8PYC9QV4.com.daimo.dev"
     ]
   },
   "webcredentials": {
     "apps": [
-      "JV8PYC9QV4.com.daimo"
+      "JV8PYC9QV4.com.daimo",
+      "JV8PYC9QV4.com.daimo.dev"
```
Comment (dcposch) [inline apps/daimo-web/public/.well-known/apple-app-site-association:28]: If I'm understanding right

This means that daimo.xyz links will open in *either* the dev app or the production app

If you have both installed, does it prompt you?

**Review by dcposch — APPROVED** [review]
LGTM

File: apps/daimo-web/public/.well-known/apple-app-site-association (hunk 2 above)
Comment (nalinbhardwaj) [reply apps/daimo-web/public/.well-known/apple-app-site-association:28]: It picks the one higher up in the list if multiple are installed: https://developer.apple.com/forums/thread/649423

**Pull request merged**